pennylane/qchem/openfermion_obs.py

```diff
@@ -649,6 +658,193 @@ def dipole_of(
+    >>> dipole_obs[0] # x-component of D
+    (
+        0.4781123173263876 * Z(0)
+      + 0.4781123173263876 * Z(1)
+      + -0.3913638489489803 * (Y(0) @ Z(1) @ Y(2))
+      + -0.3913638489489803 * (X(0) @ Z(1) @ X(2))
+      + -0.3913638489489803 * (Y(1) @ Z(2) @ Y(3))
+      + -0.3913638489489803 * (X(1) @ Z(2) @ X(3))
+      + 0.2661114704527088 * (Y(0) @ Z(1) @ Z(2) @ Z(3) @ Y(4))
+      + 0.2661114704527088 * (X(0) @ Z(1) @ Z(2) @ Z(3) @ X(4))
+      + 0.2661114704527088 * (Y(1) @ Z(2) @ Z(3) @ Z(4) @ Y(5))
+      + 0.2661114704527088 * (X(1) @ Z(2) @ Z(3) @ Z(4) @ X(5))
+      + 0.7144779061810713 * Z(2)
+      + 0.7144779061810713 * Z(3)
+      + -0.11734958781031017 * (Y(2) @ Z(3) @ Y(4))
+      + -0.11734958781031017 * (X(2) @ Z(3) @ X(4))
+      + -0.11734958781031017 * (Y(3) @ Z(4) @ Y(5))
+      + -0.11734958781031017 * (X(3) @ Z(4) @ X(5))
+      + 0.24190977644645698 * Z(4)
+      + 0.24190977644645698 * Z(5)
+    )
+    """
+
+    if method not in ["dhf", "openfermion"]:
+        raise ValueError("Only 'dhf', and 'openfermion' backends are supported.")
+
+    if mapping.strip().lower() not in ["jordan_wigner", "parity", "bravyi_kitaev"]:
+        raise ValueError(
+            f"'{mapping}' is not supported."
+            f"Please set the mapping to 'jordan_wigner', 'parity' or 'bravyi_kitaev'."
+        )
+
+    symbols = molecule.symbols
+    coordinates = molecule.coordinates
+
+    if qml.math.shape(coordinates)[0] == len(symbols) * 3:
+        geometry_dhf = qml.numpy.array(coordinates.reshape(len(symbols), 3))
+        geometry_hf = coordinates
+    elif len(coordinates) == len(symbols):
+        geometry_dhf = qml.numpy.array(coordinates)
+        geometry_hf = coordinates.flatten()
+
+    if molecule.mult != 1:
+        raise ValueError(
+            "Open-shell systems are not supported. Change the charge or spin multiplicity of the molecule."
+        )
+
+    core, active = qml.qchem.active_space(
+        molecule.n_electrons, molecule.n_orbitals, molecule.mult, active_electrons, active_orbitals
+    )
+
+    if method == "dhf":
+
+        if args is None and isinstance(geometry_dhf, qml.numpy.tensor):
+            geometry_dhf.requires_grad = False
+        mol = qml.qchem.Molecule(
+            symbols,
+            geometry_dhf,
+            charge=molecule.charge,
+            mult=molecule.mult,
+            basis_name=molecule.basis_name,
+            load_data=molecule.load_data,
+            alpha=molecule.alpha,
+            coeff=molecule.coeff,
+        )
+
+        requires_grad = args is not None
+        dip = (
+            qml.qchem.dipole_moment(mol, cutoff=cutoff, core=core, active=active, mapping=mapping)(
+                *(args or [])
+            )
+            if requires_grad
+            else qml.qchem.dipole_moment(
+                mol, cutoff=cutoff, core=core, active=active, mapping=mapping
+            )()
+        )
+        wires_map = None
```

Comment:
```suggestion
```
Why do we need to define this beforehand if we don't use it except within the `if` context?